data/data_request.py

```diff
@@ -0,0 +1,162 @@
+import requests
+from datetime import datetime
+import pandas as pd
+
+urls = {
+    "coingecko": "https://api.coingecko.com/api/v3/coins/",
+}
+
+#TODO: Give each token a quote currency already and request prices and returns in this class.
+# In the Token_Pair class, check if both token habe the same quote currency, else fail.
+# If they do have the same quote currency, pair returns can be computed as ((1+r(A)) * (1+r(B)))
+class Token():
+    """Class that represents a token"""
+    def __init__(self, name:str, ticker: str):
+        """Initializing the token object.
+
+        Args:
+            name (str): Name of the token which is used to identify the token in the data query so this must match the name of the token for the specific source.
+                NOTE: The name can be different when it should be the quote currency (e.g. 'bitcoin' becomes 'btc' if it should be the quote currency instead of the base currency)
```

Comment:
Can the ticker not be used in the quote currency instead? I don't think the caller should be aware of this convention - adding another field for naming the quote would be another solution.

Reply:
Totally! I already changed that but forgot to remove the comment here. I'm not sure actually if that works in all the cases. Do you have any idea how to confirm this? Do I need to run unit tests for every possible combination of name&ticker?